sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -439,12 +439,12 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * `Database` can be found.
    */
   override def getDatabase(dbName: String): Database = {
-    val namespace = resolveNamespace(dbName)
-    val plan = UnresolvedNamespace(namespace)
+    val plan = DescribeNamespace(UnresolvedNamespace(resolveNamespace(dbName)), false)
```

Comment:
why it's necessary to wrap `UnresolvedNamespace` with `DescribeNamespace`?

Reply:
oh, is it because `UnresolvedNamespace` have special resolution logic depending on where it is?

Reply:
Yes, [spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala](https://github.com/apache/spark/blob/9c21238ce0e6bec676213f4bad4bc1bf4ac16932/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala#L67-L68) may not be sufficient for representing the getDatabase method